gen/generator.go

```diff
@@ -567,8 +571,12 @@ func main() {
 		providerConfig["Generic"] = append(providerConfig["Generic"], genericConfigs[i].Name)
 	}
 
+	configs := make(map[string][]YamlConfig)
```

Comment:
I guess we can either create the data structure here (as you did) and then remove the `providerConfig` variable being constructed above, or we update the `providerConfig` structure to not only include the `name` but rather the full config/definition and use it instead of `configs`.

Reply:
Good point! Didnt realize the other was not used. I stuck with config for now but moved the decalrations and assignemnts around